src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs

```diff
@@ -69,7 +69,13 @@ public ProgressBar()
 
 			SizeChanged += (snd, evt) => OnSizeChange();
 
+#if !UNO_HAS_ENHANCED_LIFECYCLE
+			// Uno-specific: TODO: Investigate why we need this. It's a quite very old workaround from 2020
+			// https://github.com/unoplatform/uno/commit/641bbc9483f33c64d5eddc474f069c07b79039ba
+			// So, maybe it's no longer needed.
+			// For now, we're sure it's no longer needed with enhanced lifecycle (actually, it's problematic if it exists there)
```

Comment:
Here: https://github.com/unoplatform/uno/blob/2dbe845645483db4416d0ab9f5eedb5998ab456d/src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/ProgressBar/WinUIProgressBarPage.xaml.cs#L48

This sample listens to visual state change and updates the text

So, the ProgressBar goes into Indeterminate, causing TextBlock.Text to change which will cause LayoutUpdated, then `OnSizeChange` will cause a visual state to change from Indeterminate to Updating, etc. So, you go into a loop keeping dancing between Indeterminate and Updating states

Reply:
And it's probably worse than just the "cycle". It will really mean that any change anywhere in the visual tree will call `OnSizeChange` which will cause visual state transition to Updating that will likely mess up with ProgressBar and cause it to flicker.

Reply:
Would be great to have few details about this in the comment :)

Reply:
I extended the comment so it's clearer what the problem is.